pcbdraw/populate.py

```diff
@@ -31,9 +31,28 @@
 PKG_BASE = os.path.dirname(__file__)
 
 def parse_pcbdraw(lexer: Any, m: re.Match[str], state: Any=None) -> Any:
+    """The rule given to mistune to parse out the desired variables (side and component)"""
     text = m.group(1)
     side, components = text.split("|")
     components = list(map(lambda x: x.strip(), components.split(",")))
+    for i in reversed(range(len(components))):
+        c = components[i]
+        if c.count('-') == 1:   # if we have a - for a range (for example R3-R9)
+            m = re.match(r'(\w*?)(\d+)-(\w*?)(\d+)$', c)
+            try:
+                prefix = m.group(1)
+                if prefix != m.group(3):    # if the first and second prefix don't match, ignore
+                    continue
+                start_n = int(m.group(2))
+                end_n = int(m.group(4))
+            except (IndexError, ValueError):
+                # if we either didn't have a full regex match, or the numbers weren't integers somehow?
+                continue
+            if start_n > end_n:     # check that the first number is the lower limit (R6-R2 is not valid)
+                continue
+            # Replace XY-XZ with [XY, X(Y+1), X(Y+2), ..., X(Z-1), ZX]
+            components += [f"{prefix}{i}" for i in range(start_n, end_n+1)]
+            components.pop(i)
```

Comment:
Just an idea - wouldn't it be cleaner to modify PcbDraw to accept ranges as well for filtering and highlighting? We would cover another use case.

Reply:
yeah that's fair. I'll look into incorporating it directly in plot